kratos/utilities/math_utils.h

```diff
@@ -475,154 +717,220 @@ class MathUtils
+    
+    /**
+     * Identical but it assumes that the output vector is given already the correct size
+     * @param a: First input vector
+     * @param b: Second input vector
+     * @return c: The resulting vector
+     */
+    
+    static inline void CrossProduct(
+        array_1d<double, 3>& c, 
+        const array_1d<double, 3>& a, 
+        const array_1d<double, 3>& b
+        )
     {
         c[0] = a[1]*b[2] - a[2]*b[1];
         c[1] = a[2]*b[0] - a[0]*b[2];
         c[2] = a[0]*b[1] - a[1]*b[0];
     }
-
-    //***********************************************************************
+    
     /**
-    returns a matrix :
-        A = a.tensorproduct.b
-        a,b are assumed to be of order 3, no check is performed on the size of the vectors
-             */
-    //***********************************************************************
-    static inline MatrixType TensorProduct3(Vector& a, Vector& b)
+     * Returns a matrix :
+     * A = a.tensorproduct.b
+     * a,b are assumed to be of order 3, no check is performed on the size of the vectors
+     * @param a: First input vector
+     * @param b: Second input vector
+     */
+    
+    static inline MatrixType TensorProduct3(
+        Vector& a, 
+        Vector& b
+        )
     {
         MatrixType A(3,3);
-        A(0,0)=a[0]*b[0];
-        A(0,1)=a[0]*b[1];
-        A(0,2)=a[0]*b[2];
-        A(1,0)=a[1]*b[0];
-        A(1,1)=a[1]*b[1];
-        A(1,2)=a[1]*b[2];
-        A(2,0)=a[2]*b[0];
-        A(2,1)=a[2]*b[1];
-        A(2,2)=a[2]*b[2];
+        
+        A(0,0) = a[0]*b[0];
+        A(0,1) = a[0]*b[1];
+        A(0,2) = a[0]*b[2];
+        A(1,0) = a[1]*b[0];
+        A(1,1) = a[1]*b[1];
+        A(1,2) = a[1]*b[2];
+        A(2,0) = a[2]*b[0];
+        A(2,1) = a[2]*b[1];
+        A(2,2) = a[2]*b[2];
+        
         return A;
     }
-
-    //***********************************************************************
+    
     /**
-        "InputMatrix" is ADDED to "Destination" matrix starting from
-        InitialRow and InitialCol of the destination matrix
-        "Destination" is assumed to be able to contain the "input matrix"
-        (no check is performed on the bounds)
-             */
-    //***********************************************************************
+     * "InputMatrix" is ADDED to "Destination" matrix starting from
+     * InitialRow and InitialCol of the destination matrix
+     * "Destination" is assumed to be able to contain the "input matrix"
+     * (no check is performed on the bounds)
+     * @return Destination: The matric destination
+     * @param InputMatrix: The input matrix to be computed
+     * @param InitialRow: The initial row to compute
+     * @param InitialCol: The initial column to compute
+     */
+    
     static inline void  AddMatrix(
         MatrixType& Destination,
         MatrixType& InputMatrix,
         int InitialRow,
```

Comment:
you could make const the InputMatrix and the integer indices (this one is only a suggestion)